code/modules/client/client_procs.dm

```diff
@@ -529,17 +529,18 @@ var/global/const/MAX_VIEW = 41
 	if(mob)
 		mob.reload_fullscreen()
 
-/client/proc/toggle_fullscreen(new_value)
-	if((new_value == PREF_BASIC) || (new_value == PREF_FULL))
-		winset(src, "mainwindow", "is-maximized=false;can-resize=false;titlebar=false")
-		if(new_value == PREF_FULL)
-			winset(src, "mainwindow", "menu=null;statusbar=false")
-		winset(src, "mainwindow.split", "pos=0x0")
-	else
-		winset(src, "mainwindow", "is-maximized=false;can-resize=true;titlebar=true")
-		winset(src, "mainwindow", "menu=menu;statusbar=true")
-		winset(src, "mainwindow.split", "pos=3x0")
-	winset(src, "mainwindow", "is-maximized=true")
+/client/proc/toggle_fullscreen(value)
+	set waitfor = FALSE
+
+	winset(src, null, {"
+	mainwindow.is-maximized = false;
+	mainwindow.can-resize = [(value == PREF_BASIC) || (value == PREF_FULL) ? "false" : "true"];
+	mainwindow.titlebar = [(value == PREF_BASIC) || (value == PREF_FULL) ? "false" : "true"];
+	mainwindow.menu = [value == PREF_FULL ? "null" : "menu"];
+	mainwindow.statusbar = [value == PREF_FULL ? "false" : "true"];
+	mainwindow.split.pos = [(value == PREF_BASIC) || (value == PREF_FULL) ? "0x0" : "3x0"];
+	"})
+	winset(src, null, "mainwindow.is-maximized = true;")
```

Comment:
Why batch all of them other than this one?

Reply:
Because byond issue: proper updating can be done only if you change is-maximized after other parameters have been set.